### ISSUE TITLE
ComboPointWidget causes Nil value exception, WoW client stops responding

### ISSUE THREAD

**Issue by malchwo** [Luxocracy/NeatPlates#263]
Hi,

I'm loving the addon as a superb replacement for the default nameplates. But recently I'm running into a problem when playing Brewmaster-Monk. I'm not quite sure when the problem started, but I don't think it was an issue before shadowlands. 

When playing Brewmaster, a spec without combo point mechanic, the ComboPointWidget.lua causes a nil value exception. While the exception infrequently occurs in fights outdoors/maw/thorgast or when attacking a training dummy, it repeatable and frequently occurs in dungeons (most of the time within the first few pulls).
Due to the massive amount of repeated errors the WoW client stops responding when no BugGrabber-Addon is used. 

Fresh install, no changes in the default settings, it doesn't matter if the char was logged in with Brewmaster spec or switched from Windwalker to Brewmaster in a session.
The exception doesn't occur while playing Windwalker itself (which has combo points as a mechanic), ~~not tested with Mistweaver~~ but happens with Mistweaver as well (also: no combo points).

Thanks for the great work.

``` lua
352x NeatPlatesWidgets\ComboPointWidget.lua:169: attempt to perform arithmetic on local 'offset' (a nil value)
[string "@NeatPlatesWidgets\ComboPointWidget.lua"]:169: in function <NeatPlatesWidgets\ComboPointWidget.lua:138>
[string "@NeatPlatesWidgets\ComboPointWidget.lua"]:245: in function <NeatPlatesWidgets\ComboPointWidget.lua:241>

Locals:
frame = <unnamed> {
 0 = <userdata>
 UpdateScale = <function> defined @NeatPlatesWidgets\ComboPointWidget.lua:182
 Update = <function> defined @NeatPlatesWidgets\ComboPointWidget.lua:138
 PartialFill = <unnamed> {
 }
 _Hide = <function> defined =[C]:-1
 guid = "Creature-0-3103-2284-12420-171448-0000439D78"
 UpdateContext = <function> defined @NeatPlatesWidgets\ComboPointWidget.lua:203
 Icon = <unnamed> {
 }
 Hide = <function> defined @NeatPlatesWidgets\ComboPointWidget.lua:351
}
points = 1
maxPoints = 4
pattern = <table> {
 w = 80
 h = 20
}
offset = nil
(*temporary) = <function> defined =[C]:-1
(*temporary) = <unnamed> {
 0 = <userdata>
}
(*temporary) = nil
(*temporary) = nil
(*temporary) = 0.062500
(*temporary) = nil
(*temporary) = 1
(*temporary) = <table> {
 UpdateOnBarHighlightMarksBySpell = <function> defined @FrameXML\ActionButton.lua:78
 ERR_OUT_OF_CHI = "Not enough chi"
 DH_HAVOC_CORE_ABILITY_2 = "Strong melee attack that consumes Fury. If it critical strikes, some Fury is refunded."
 MultiCastActionButton6Cooldown = MultiCastActionButton6Cooldown {
 }
 MerchantItem9ItemButtonStock = MerchantItem9ItemButtonStock {
 }
 GetTrainerServiceTypeFilter = <function> defined =[C]:-1
 UNIT_NAMES_COMBATLOG_TOOLTIP = "Color unit names."
 SetTrainerServiceTypeFilter = <function> defined =[C]:-1
 LE_GAME_ERR_CHAT_RAID_RESTRICTED_TRIAL = 743
 SPELL_FAILED_CUSTOM_ERROR_71 = "This partygoer wants to dance with you."
 LE_GAME_ERR_PET_SPELL_TARGETS_DEAD = 401
 ERROR_CLUB_TICKET_COUNT_AT_MAX_COMMUNITY = "Can't create any more invite links for this community."
 RecruitAFriendFrame = RecruitAFriendFrame {
 }
 CompactUnitFrameProfilesGeneralOptionsFrameHealthTextDropdownButtonNormalTexture = CompactUnitFrameProfilesGeneralOptionsFrameHealthTextDropdownButtonNormalTexture {
 }
 TutorialFrameLeft19 = TutorialFrameLeft19 {
 }
 MultiCastActionButton2Cooldown = MultiCastActionButton2Cooldown {
 }
 ERR_TRADE_EQUIPPED_BAG = "You can't trade equipped bags."
 PVP_RANK_6_1 = "Corporal"
 MultiBarLeftButton7 = MultiBarLeftButton7 {
 }
 InterfaceOptionsNamesPanelUnitNameplatesShowAll = InterfaceOptionsNamesPanelUnitNameplatesShowAll {
 }
 VideoOptionsFrameDefaults = VideoOptionsFrameDefaults {
 }
 MerchantItem1AltCurrencyFrameItem1Text = MerchantItem1AltCurrencyFrameItem1Text {
 }
 OPTION_TOOLTIP_ACTION_BUTTON_USE_KEY_DOWN = "Action button keybinds will respond on key down, rather than on key up."
 BINDING_NAME_NAMEPLATES = "Show Enemy Name Plates"
 CHAT_HEADER_SUFFIX = ": "
 MultiBarBottomRightButton8Shine5 = MultiBarBottomRightButton8Shine5 {
 }
 COVENANT_MISSIONS_HEALTH = "Health"
 MAIL_LETTER_TOOLTIP = "Click to make a permanent
copy of this letter."
 UnitFrameManaBar_UnregisterDefaultEvents = <function> defined @FrameXML\UnitFrame.lua:789
 PaperDollTitlesPaneButton10BgTop = PaperDollTitlesPaneButton10BgTop {
 }
 ERR_NOAMMO_S = "%s"
 CHAT_CONFIG_OTHER_COMBAT = <table> {
 }
 FCFDockOverflowButton_OnClick = <function> defined @FrameXML\FloatingChatFrame.lua:2379
 TutorialFrameRight19 = TutorialFrameRight19 {
 }
 ChatFrame6EditBoxFocusMid = ChatFrame6EditBoxFocusMid {
 }
 BN_UNABLE_TO_RESOLVE_NAME = "Unable to whisper '%s'. Blizzard services may be unavailable."
 AutoCompleteEditBox_OnKeyDown = <function> defined @FrameXML\AutoComplete.lua:366
 CompactRaidFrameManagerDisplayFrameHiddenModeToggleTopRight = CompactRaidFrameManagerDisplayFrameHiddenModeToggleTopRight {
 }
 LFGTeleport = <function> defined =[C]:-1
 LE_GAME_ERR_ONLY_ONE_QUIVER = 33
 INT_SPELL_DURATION_HOURS = "%d |4hour:hrs;"
 SLASH_LibQTip1 = "/qtip"
 ToggleEncounterJournal = <function> defined @FrameXML\UIParent.lua:1039
 LOSS_OF_CONTROL_DISPLAY_FEAR = "Feared"
 Graphics_QualityText = Graphics_QualityText {
 }
 OverrideActionBarButton6Shine11 = OverrideActionBarButton6Shine11 {
 }
 ROGUE_COMBAT_CORE_ABILITY_4 = "Melee attack that spends Combo Points."
 JoinSkirmish = <function> defined =[C]:-1
 FriendsFriendsScrollFrameButton14 = FriendsFriendsScrollFrameButton14 {
 }
```

**Comment by Luxocracy** [comment]
I'll look into asap, until then if it's urgent you can make a profile for these specs with the combo point widget disabled.

**Comment by Luxocracy** [comment]
I wasn't able to replicate this on my monk, probably because he isn't max level, but I think this should be fixed in the latest release 9.0.2.17.

**Comment by malchwo** [comment]
Just tried the updated version, works like a charm. Thanks a lot for the quick fix on a weekend!
